app/controllers/OverviewController.scala

```diff
@@ -233,29 +253,43 @@ class OverviewController @Inject()(
         .addingToSession((Constants.SRN, srn.value))
     }
 
-  def onSelectStart(srn: Srn, taxYear: String, version: String, reportType: String): Action[AnyContent] =
-    identify.andThen(allowAccess(srn)).andThen(getData).andThen(createData).async { implicit request =>
-      reportType match {
-        case PsrReportType.Sipp.name =>
-          val sippUrl = s"${config.urls.sippBaseUrl}/${srn.value}${config.urls.sippStartJourney}"
-          Future.successful(
-            Redirect(sippUrl)
-              .addingToSession(Constants.TAX_YEAR -> taxYear)
-              .addingToSession(Constants.VERSION -> version)
-          )
-        case _ =>
-          val yearFrom = LocalDate.parse(taxYear)
-          val yearTo = yearFrom.plusYears(1).minusDays(1)
-          val dateRange = DateRange(yearFrom, yearTo)
-          for {
-            userAnswers <- Future.fromTry(request.userAnswers.set(WhichTaxYearPage(srn), dateRange))
-            _ <- saveService.save(userAnswers.copy(id = UNCHANGED_SESSION_PREFIX + userAnswers.id))
-            _ <- saveService.save(userAnswers)
-          } yield {
-            Redirect(controllers.routes.WhatYouWillNeedController.onPageLoad(srn, "", taxYear, version))
-          }
+  def onSelectStart(
+    srn: Srn,
+    taxYear: String,
+    version: String,
+    reportType: String,
+    lastSubmittedPsrFbInPreviousYears: Option[String]
+  ): Action[AnyContent] =
+    identify
+      .andThen(allowAccess(srn))
+      .andThen(createData)
+      .andThen(
+        prePopData(
+          Option.when(reportType == PsrReportType.Standard.name)(lastSubmittedPsrFbInPreviousYears).flatten
+        )
+      )
+      .async { implicit request =>
+        reportType match {
+          case PsrReportType.Sipp.name =>
+            val sippUrl = s"${config.urls.sippBaseUrl}/${srn.value}${config.urls.sippStartJourney}"
+            Future.successful(
+              Redirect(sippUrl)
+                .addingToSession(Constants.TAX_YEAR -> taxYear)
+                .addingToSession(Constants.VERSION -> version)
+            )
+          case _ =>
+            val yearFrom = LocalDate.parse(taxYear)
+            val yearTo = yearFrom.plusYears(1).minusDays(1)
+            val dateRange = DateRange(yearFrom, yearTo)
+            for {
+              userAnswers <- Future.fromTry(request.userAnswers.set(WhichTaxYearPage(srn), dateRange))
+              _ <- saveService.save(userAnswers.copy(id = UNCHANGED_SESSION_PREFIX + userAnswers.id))
```

Comment:
Maybe I misunderstood something, but I'm struggling to find where does `request.prePopUserAnswers` got copied over to request.userAnswers. Should this happen here?

Reply:
It's not copied over yet. After cleaning up either we can copy `request.prePopUserAnswers` into currentUA and (delete `request.prePopUserAnswers` or keep it here )